appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java

```diff
@@ -208,95 +126,31 @@ public void updateMenuState(int numSelectedProjects, int numProjects) {
 
-  private void createProjectsMenu() {
-    List<DropDownItem> fileItems = Lists.newArrayList();
-    fileItems.add(new DropDownItem(WIDGET_NAME_MY_PROJECTS, MESSAGES.projectMenuItem(),
-        new SwitchToProjectAction()));
-    fileItems.add(null);
-    if (!isReadOnly) {
-      fileItems.add(new DropDownItem(WIDGET_NAME_NEW, MESSAGES.newProjectMenuItem(),
-          new NewAction()));
-      fileItems.add(new DropDownItem(WIDGET_NAME_IMPORTPROJECT, MESSAGES.importProjectMenuItem(),
-          new ImportProjectAction()));
-      fileItems.add(new DropDownItem(WIDGET_NAME_IMPORTTEMPLATE, MESSAGES.importTemplateButton(),
-          new ImportTemplateAction()));
-      fileItems.add(new DropDownItem(WIDGET_NAME_DELETE, MESSAGES.deleteProjectButton(),
-          new DeleteAction()));
-      fileItems.add(new DropDownItem(WIDGET_NAME_DELETE_TRASH, MESSAGES.deleteFromTrashButton(),
-          new DeleteForeverProjectAction()));
-      fileItems.add(null);
-      fileItems.add(new DropDownItem(WIDGET_NAME_SAVE, MESSAGES.saveMenuItem(),
-          new SaveAction()));
-      fileItems.add(new DropDownItem(WIDGET_NAME_SAVE_AS, MESSAGES.saveAsMenuItem(),
-          new SaveAsAction()));
-      fileItems.add(new DropDownItem(WIDGET_NAME_CHECKPOINT, MESSAGES.checkpointMenuItem(),
-          new CheckpointAction()));
-      fileItems.add(null);
-    }
-    fileItems.add(new DropDownItem(WIDGET_NAME_EXPORTPROJECT, MESSAGES.exportProjectMenuItem(),
-        new ExportProjectAction()));
-    fileItems.add(new DropDownItem(WIDGET_NAME_EXPORTALLPROJECTS, MESSAGES.exportAllProjectsMenuItem(),
-        new ExportAllProjectsAction()));
-    fileItems.add(null);
-    if (!isReadOnly) {
-      fileItems.add(new DropDownItem(WIDGET_NAME_UPLOAD_KEYSTORE, MESSAGES.uploadKeystoreMenuItem(),
-          new UploadKeystoreAction()));
-    }
-    fileItems.add(new DropDownItem(WIDGET_NAME_DOWNLOAD_KEYSTORE, MESSAGES.downloadKeystoreMenuItem(),
-        new DownloadKeystoreAction()));
-    if (!isReadOnly) {
-      fileItems.add(new DropDownItem(WIDGET_NAME_DELETE_KEYSTORE, MESSAGES.deleteKeystoreMenuItem(),
-          new DeleteKeystoreAction()));
-    }
-    refreshMenu(fileDropDown, fileItems);
-  }
+  public TopToolbar() {
+    // Should the UI be in read only mode?
+    isReadOnly = Ode.getInstance().isReadOnly();
 
-  private void createConnectMenu() {
-    List<DropDownItem> connectItems = Lists.newArrayList();
-    connectItems.add(new DropDownItem(WIDGET_NAME_WIRELESS_BUTTON,
-        MESSAGES.AICompanionMenuItem(), new WirelessAction()));
-    if (iamChromebook) {
-      connectItems.add(new DropDownItem(WIDGET_NAME_CHROMEBOOK,
-          MESSAGES.chromebookMenuItem(), new ChromebookAction()));
-    } else {
-      connectItems.add(new DropDownItem(WIDGET_NAME_EMULATOR_BUTTON,
-          MESSAGES.emulatorMenuItem(), new EmulatorAction()));
-      connectItems.add(new DropDownItem(WIDGET_NAME_USB_BUTTON, MESSAGES.usbMenuItem(),
-          new UsbAction()));
-    }
```

Comment:
Note that the Chromebook logic has been lost here. We probably want to have it in the ui.xml file and then show either the Chromebook or [Emulator, Usb] options depending on the platform.